python/lsst/ip/diffim/subtractImages.py

```diff
@@ -642,23 +656,42 @@ def _subtractImages(science, template, backgroundModel=None):
     return difference
 
 
-def _shapeTest(psf1, psf2):
+def _shapeTest(exp1, exp2, fwhmExposureBuffer, fwhmExposureGrid):
     """Determine whether psf1 is narrower in either dimension than psf2.
 
     Parameters
     ----------
-    psf1 : `lsst.afw.detection.Psf`
-        Reference point spread function (PSF) to evaluate.
-    psf2 : `lsst.afw.detection.Psf`
-        Candidate point spread function (PSF) to evaluate.
-
+    exp1 : `~lsst.afw.image.Exposure`
+        Exposure with the reference point spread function (PSF) to evaluate.
+    exp2 : `~lsst.afw.image.Exposure`
+        Exposure with a candidate point spread function (PSF) to evaluate.
+    fwhmExposureBuffer : `float`
+        Fractional buffer margin to be left out of all sides of the image
+        during the construction of the grid to compute mean PSF FWHM in an
+        exposure, if the PSF is not available at its average position.
+    fwhmExposureGrid : `int`
+        Grid size to compute the mean FWHM in an exposure, if the PSF is not
+        available at its average position.
     Returns
     -------
-    `bool`
-        Returns True if psf1 is narrower than psf2 in either dimension.
+    result : `bool`
+        True if ``exp1`` has a narrower PSF than ``exp2`` in either dimension.
     """
-    shape1 = getPsfFwhm(psf1, average=False)
-    shape2 = getPsfFwhm(psf2, average=False)
+    try:
+        shape1 = getPsfFwhm(exp1.psf, average=False)
+        shape2 = getPsfFwhm(exp2.psf, average=False)
+    except InvalidParameterError:
+        shape1 = evaluateMeanPsfFwhm(exp1,
+                                     fwhmExposureBuffer=fwhmExposureBuffer,
+                                     fwhmExposureGrid=fwhmExposureGrid
+                                     )
+        shape2 = evaluateMeanPsfFwhm(exp2,
+                                     fwhmExposureBuffer=fwhmExposureBuffer,
+                                     fwhmExposureGrid=fwhmExposureGrid
+                                     )
+        return shape1 < shape2
```

Comment:
Minor point, but this should probably be `<=`. `xTest` and `yTest` below should probably use `<=` as well, which was a mistake in the original implementation.

Reply:
Done. And I've updated the docstring also accordingly.